dsp/window/window.go

```diff
@@ -92,14 +92,14 @@ func Triangular(seq []float64) []float64 {
 // The Hann window is a high-resolution window.
 //
 // The sequence weights are
-//  w[k] = 0.5*(1 - cos(2*π*(k+1/2)/N)),
+//  w[k] = 0.5*(1 - cos(2*π*k/(N-1))),
 // for k=0,1,...,N-1 where N is the length of the window.
 //
 // Spectral leakage parameters: ΔF_0 = 4, ΔF_0.5 = 1.5, K = 2, ɣ_max = -31.5, β = -6.
 func Hann(seq []float64) []float64 {
-	k := 2 * math.Pi / float64(len(seq))
+	k := 2 * math.Pi / float64(len(seq)-1)
 	for i := range seq {
-		seq[i] *= 0.5 * (1 - math.Cos(k*(float64(i)+0.5)))
+		seq[i] *= 0.5 * (1 - math.Cos(k*float64(i)))
```

Comment:
I'm slightly worried that, numerically, `(2 * math.Pi / float64(len(seq) - 1)) * float64(i) != 2 * math.Pi` for `i == len(seq) - 1`, and we lose the symmetry at the edges. A slightly more expensive but maybe more accurate way of computing this would be `2 * math.Pi * (float64(i) / float64(len(seq) - 1))`. WTDYT?

Reply:
Do you have an estimate of the impact or examples where it presents itself? Ideally, this commit would be a pure reversion (it's not because other code has changed, but it's fairly close), so I'd be happy to look at it, but probably after this has settled and also since Vladimir wants to look into it more before the review goes much further (the non-reversion parts of this PR are worth looking at though).

Reply:
I did a quick test
```
package main

import (
	"fmt"
	"math"
)

func main() {
	n := 75
	k := 2 * math.Pi / float64(n - 1)
	a := math.Cos(k * 0)
	b := math.Cos(k * float64(n - 1))
	fmt.Printf("a - b == %g", a - b)
}
```
and it doesn't seem to be a problem. Sorry for jumping the gun.